src/main/scala/org/ergoplatform/settings/ErgoValidationSettings.scala

```diff
@@ -65,10 +74,11 @@ case class ErgoValidationSettings(rules: Map[Short, RuleStatus]) extends Validat
 }
 
 object ErgoValidationSettings {
-
   /**
     * Checks that s1 and s2 are equals
     */
+  // TODO optimize: throwing exception by default incur building of a stack trace, which is 10-100x
```

Comment:
There is a task for this kind of optimizations https://github.com/ergoplatform/ergo/issues/746, no need to put TODOs. Also I propose to sut-up benchmarks and check first, that this optimization will really improve performance, rather than just increase code complexity without real performance improvement.

Reply:
heh, I put todo first, and than reported this problem :)
I definitely don't want to argue.